corehq/apps/users/templates/users/partials/edit_role_modal.html

```diff
@@ -20,13 +20,14 @@ <h4 class="modal-title" data-bind="text: modalTitle"></h4>
           <div class="form-group">
             <label class="col-sm-4 control-label"
                    for="role-name">
-              {% trans "Role Name" %}
+              {% trans "Role Name*" %}
             </label>
             <div class="col-sm-8 controls">
               <input data-bind="value: name, disable: !$root.allowEdit"
                      type="text"
                      id="role-name"
-                     class="form-control" />
+                     class="form-control"
+                     required />
```

Comment:
Ah, I think I see why this wasn't working - it isn't in the `<form />` block.  Can you see what happens if you extend the form element to encompass all of these fields?  (Conceptually, they _are_ all part of the form)

Reply:
I just tried that, and it didn't seem to help (never highlighted the input box as red). I think this is because of the issue we discussed before, where the `enable: allowSubmit` means that the button is unable to submit unless there is an input, rendering the `required` flag unnecessary.
The mobile workers page works the same way, where they just added an asterisk next to the required inputs but otherwise don't highlight to the user what inputs are needed.

Reply:
Did you remove the `novalidate` when you tried that?  I just mocked it up locally, and this is what I see when I click "Save" with the name field empty:
![image](https://user-images.githubusercontent.com/2367539/84941979-2897ef80-b0b0-11ea-9f8d-a68cd28ae5b2.png)

Reply:
Ah no I forgot about the `novalidate`. I'll try it again.

Reply:
Hopefully that should make it obvious to the user what the issue is, and it will probably make `allowSubmit` unnecessary.

Reply:
In your local environment, you don't have the same `enable: allowSubmit` on the save button, right? Because even after creating an all-encompassing form and removing `novalidate` I still can't click the button when there is no name input.

I'm not sure how the `required` approach would fit in alongside the current `allowSubmit` functionality.